### PR TITLE
Fix missing blackboard bold letters

### LINKED ISSUE #3725: Missing blackboard bold letters
### Description

While combing through unicode, I discovered that there are some exceptional blackboard bold symbols that are currently inaccessible in typst (short of manually entering the codepoint). Explicitly:

Blackboard bold:

Greek uppercase: gamma (213E), pi (213F)
Greek lowercase: gamma (213D), pi (213C)
Summation symbol (2140)

Italic + blackboard bold:

Latin lowercase: d (2146), e (2147), i (2148), j (2149)
Latin uppercase: d (2145)

### Reproduction URL

_No response_

### Operating system

_No response_

### Typst version

- [X] I am using the latest version of Typst

### PR DESCRIPTION
**Pull request by joserlopes**
Fixes #3725

# Explanation

Some exceptional blackboard bold symbols were currently inaccessible in typst, namely: 

Blackboard bold:

Greek uppercase: gamma (213E), pi (213F)
Greek lowercase: gamma (213D), pi (213C)
Summation symbol (2140)

Italic + blackboard bold:

Latin lowercase: d (2146), e (2147), i (2148), j (2149)
Latin uppercase: d (2145)

# What was done

- Added auto italic to math symbols only if they weren't blackboard styled.
- Manually added all the missing exceptions to `style.rs` inside the `math` folder.
- The test `style.typ` inside the `math` test folder and the corresponding image were updated to reflect the new changes.

### CODE REVIEW

**Comment by joserlopes** [comment]
Broken tests due to unrelated issue.

File: crates/typst/src/math/style.rs [hunk 1]
```diff
@@ -300,7 +300,8 @@ pub fn styled_char(styles: StyleChain, c: char, auto_italic: bool) -> char {
                 c,
                 'a'..='z' | 'ı' | 'ȷ' | 'A'..='Z' | 'α'..='ω' |
                 '∂' | 'ϵ' | 'ϑ' | 'ϰ' | 'ϕ' | 'ϱ' | 'ϖ'
-            ),
+            )
+            && variant != Bb,
```
Comment (laurmaedje) [inline crates/typst/src/math/style.rs:304]: What's the reasoning for this? In isolation (for just bb), it seems a bit ad-hoc.

**Review by laurmaedje — COMMENTED** [review]
How did you determine the missing exceptions? I wonder whether we've really got them all now.

File: crates/typst/src/math/style.rs (hunk 1 above)
Comment (joserlopes) [reply crates/typst/src/math/style.rs:304]: The reason was that, if a bb symbol was automatically italized then it wouldn't enter the exception condition, therefore not being rendered properly

**Comment by joserlopes** [comment]
I just added the exceptions that were listed on the issue, there is a chance more exceptions need to be added.

**Comment by Enivex** [comment]
> How did you determine the missing exceptions? I wonder whether we've really got them all now.

There are some Arabic double struck symbols as well, but I know nothing about Arabic math or if they're already available in typst or not.

Other than those, I searched pretty thoroughly, and my list coincides with the list on Wikipedia https://en.m.wikipedia.org/wiki/Blackboard_bold

File: crates/typst/src/math/style.rs (hunk 1 above)
Comment (laurmaedje) [reply crates/typst/src/math/style.rs:304]: Maybe we should do auto-italic only for `matches!(variant, Sans | Serif)`. these are the ones that have proper italic variants for all glyphs. 

cc: @Enivex thoughts?

Comment (joserlopes) [reply crates/typst/src/math/style.rs:304]: So, would it be enough to just change that line to become `matches!(variant, Sans | Serif)`?

Comment (laurmaedje) [reply crates/typst/src/math/style.rs:304]: yeah I think that would make sense

Comment (joserlopes) [reply crates/typst/src/math/style.rs]: done!

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**